cost-analyzer/templates/grafana-datasource-template.yaml

```diff
@@ -1,3 +1,7 @@
+{{- if .Values.grafana -}}
+{{- if .Values.grafana.sidecar -}}
+{{- if .Values.grafana.datasources -}}
+{{- if .Values.grafana.datasources.enabled -}}
 apiVersion: v1
```

Comment:
Like with the dashboards, all we need for the datasource is `{{- if .Values.grafana.datasources.enabled -}}`; the other conditionals are not helpful.

Reply:
Ditto re: unnecessary conditionals. However, this does raise a point about proxying to grafana and assumptions we've made about grafana being reachable-- I've addressed those in another commit.